### PR TITLE
Fix potential file inclusion via variable error

### PR DESCRIPTION
**Pull request by soniasingla**
Signed-off-by: Sonia Singla <soniasingla.1812@gmail.com>
Bugsbash

### CODE REVIEW

**Review by akhilerm — CHANGES_REQUESTED** [review]
@soniasingla Can you run the unit-tests and make sure the tests passes for the package? I can see that there are some compilation issues for the proposed changes.

**Comment by akhilerm** [comment]
@soniasingla Can you fix the conflicts and update on the PR.

File: pkg/udev/mockdata.go [hunk 1]
```diff
@@ -140,7 +142,10 @@ func OsDiskName() (string, string, error) {
 
 // getSyspathOfOsDisk returns syspath of os disk in success
 func getSyspathOfOsDisk(osDiskName string) (string, error) {
-	data, err := ioutil.ReadFile("/sys/class/block/" + osDiskName + "/dev")
+	basePath := "/sys/class/block/"
+	filePath := "/dev"
+	filePath = path.Join(basePath, filePath.Clean(filePath))
+	data, err := ioutil.ReadFile(basePath + osDiskName + filePath)
```
Comment (richardelling) [inline pkg/udev/mockdata.go:148]: it seems clearer to me to do the path.Join and Clean inside ReadFile()

**Comment by akhilerm** [comment]
@soniasingla Can you address the review comments and rebase the PR.

File: pkg/udev/mockdata.go (hunk 1 above)
Comment (soniasingla) [reply pkg/udev/mockdata.go:148]:
```suggestion
	data, err := ioutil.ReadFile(basePath + osDiskName + path.Join(basePath, filePath.Clean(filePath)))
```

do you mean this @richardelling

Comment (richardelling) [reply pkg/udev/mockdata.go:148]: This still seems wrong to me. AIUI, the file to be read is `/sys/class/block/nvme0n1/dev` So I'm expecting something like:
```
basePath := "/sys/class/block"
filePath := "dev"
data, err := ioutil.ReadFile(filepath.Clean(path.Join(basePath, osDiskName, filePath)))
```

Comment (soniasingla) [reply pkg/udev/mockdata.go]: thanks @richardelling made the changes

Comment (akhilerm) [reply pkg/udev/mockdata.go]: @soniasingla Can you do the same set of changes in the other 3 files also?

**Comment by soniasingla** [comment]
@akhilerm I undo'ed those changes as I think now musedev is not reporting them

File: pkg/udev/mockdata.go [hunk 2]
```diff
@@ -151,7 +154,9 @@ func getSyspathOfOsDisk(osDiskName string) (string, error) {
 
 // getOsDiskSize returns size of os disk in success
 func getOsDiskSize(osDiskName string) (string, error) {
-	sizeByte, err := ioutil.ReadFile(filepath.Clean("/sys/class/block/" + osDiskName + "/size"))
+	basePath := "/sys/class/block/"
+	fileName := "/dev"
```
Comment (richardelling) [inline pkg/udev/mockdata.go:158]: for consistency with line 147 above, we can use "dev"

**Comment by akhilerm** [comment]
> I undo'ed those changes as I think now musedev is not reporting them

@soniasingla  But you can still fix them, because that issue is there.

**Review by akhilerm — APPROVED** [review]
LGTM

File: pkg/udev/mockdata.go [hunk 3]
```diff
@@ -151,7 +154,9 @@ func getSyspathOfOsDisk(osDiskName string) (string, error) {
 
 // getOsDiskSize returns size of os disk in success
 func getOsDiskSize(osDiskName string) (string, error) {
-	sizeByte, err := ioutil.ReadFile(filepath.Clean("/sys/class/block/" + osDiskName + "/size"))
+	basePath := "/sys/class/block/"
+	fileName := "dev"
```
Comment (akhilerm) [inline pkg/udev/mockdata.go:158]:
```suggestion
	fileName := "size"
```

**Review by akhilerm — CHANGES_REQUESTED** [review]
One change in file name is needed.

File: pkg/udev/mockdata.go [hunk 4]
```diff
@@ -142,7 +143,9 @@ func OsDiskName() (string, string, error) {
 
 // getSyspathOfOsDisk returns syspath of os disk in success
 func getSyspathOfOsDisk(osDiskName string) (string, error) {
-	data, err := ioutil.ReadFile(filepath.Clean("/sys/class/block/" + osDiskName + "/dev"))
+	basePath := "/sys/class/block"
+	fileName := "size"
```
Comment (akhilerm) [inline pkg/udev/mockdata.go:147]: This file is dev, 
```suggestion
	fileName := "dev"
```

**Review by akhilerm — CHANGES_REQUESTED** [review]
One file to be read is `dev` and the other is `size`.

**Review by akhilerm — APPROVED** [review]
LGTM

File: pkg/udev/mockdata.go [hunk 5]
```diff
@@ -142,7 +143,9 @@ func OsDiskName() (string, string, error) {
 
 // getSyspathOfOsDisk returns syspath of os disk in success
 func getSyspathOfOsDisk(osDiskName string) (string, error) {
-	data, err := ioutil.ReadFile(filepath.Clean("/sys/class/block/" + osDiskName + "/dev"))
+	basePath := "/sys/class/block"
```
Comment (akhilerm) [inline pkg/udev/mockdata.go:146]:
```suggestion
	basePath := "/sys/class/block/"
```

just one more comment, we can make this consistent across the file.

**Review by akhilerm — DISMISSED** [review]
LGTM

File: pkg/udev/mockdata.go [hunk 6]
```diff
@@ -142,7 +143,9 @@ func OsDiskName() (string, string, error) {
 
 // getSyspathOfOsDisk returns syspath of os disk in success
 func getSyspathOfOsDisk(osDiskName string) (string, error) {
-	data, err := ioutil.ReadFile(filepath.Clean("/sys/class/block/" + osDiskName + "/dev"))
+	basePath := "/sys/class/block/"
+	fileName := "dev"
+	data, err := ioutil.ReadFile(filepath.Clean(path.Join(basePath, osDiskName, fileName)))
```
Comment (z0marlin) [inline pkg/udev/mockdata.go:148]:
```suggestion
	data, err := ioutil.ReadFile(filepath.Clean(path.Join("/sys/class/block/", osDiskName, "dev")))
```

File: pkg/udev/mockdata.go [hunk 7]
```diff
@@ -151,7 +154,9 @@ func getSyspathOfOsDisk(osDiskName string) (string, error) {
 
 // getOsDiskSize returns size of os disk in success
 func getOsDiskSize(osDiskName string) (string, error) {
-	sizeByte, err := ioutil.ReadFile(filepath.Clean("/sys/class/block/" + osDiskName + "/size"))
+	basePath := "/sys/class/block/"
+	fileName := "size"
+	sizeByte, err := ioutil.ReadFile(filepath.Clean(path.Join(basePath, osDiskName, fileName)))
```
Comment (z0marlin) [inline pkg/udev/mockdata.go:159]:
```suggestion
	sizeByte, err := ioutil.ReadFile(filepath.Clean(path.Join("/sys/class/block/", osDiskName, "size")))
```

We don't seem to be using the variables `fileName` and `basePath` anywhere else in the function. I think we can directly use the strings  for one time use.

**Review by z0marlin — CHANGES_REQUESTED** [review]

**Comment by soniasingla** [comment]
@z0marlin updated the PR, thanks

**Review by z0marlin — APPROVED** [review]

**Review by akhilerm — APPROVED** [review]
LGTM

**Comment by soniasingla** [comment]
@akhilerm here we go

**Review by akhilerm — APPROVED** [review]
LGTM

**Review by richardelling — COMMENTED** [review]
LGTM thanks!

**Pull request merged**